src/NuGet.Clients/NuGet.VisualStudio.Internal.Contracts/ContextInfos/TransitivePackageReferenceContextInfo.cs

```diff
@@ -0,0 +1,76 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using NuGet.Frameworks;
+using NuGet.Packaging;
+using NuGet.Packaging.Core;
+using NuGet.Versioning;
+
+namespace NuGet.VisualStudio.Internal.Contracts
+{
+    public class TransitivePackageReferenceContextInfo : ITransitivePackageReferenceContextInfo
+    {
+        public TransitivePackageReferenceContextInfo(PackageIdentity identity, NuGetFramework? framework)
+        {
+            if (identity == null)
+            {
+                throw new ArgumentNullException(nameof(identity));
+            }
+            // framework is null for project.json package references. 
+
+            Identity = identity;
+            Framework = framework;
+            TransitiveOrigins = new List<IPackageReferenceContextInfo>();
+        }
+
+        internal static TransitivePackageReferenceContextInfo Create(PackageIdentity identity, NuGetFramework? framework)
```

Comment:
Maybe add a comment that indicates this is for testing only like PackageReferenceContextInfo has.

Reply:
Added.